service/matching/taskQueueManager.go

```diff
@@ -630,3 +700,71 @@ func newIOContext() (context.Context, context.CancelFunc) {
 
 	return ctx, cancel
 }
+
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartitionOnInit(ctx context.Context) {
+	if c.metadataInitialFetchFut.Ready() {
+		return
+	}
+	_, err := c.fetchMetadataFromRootPartition(ctx)
+	c.metadataInitialFetchFut.Set(struct{}{}, err)
+}
+
+// fetchMetadataFromRootPartition fetches metadata from root partition iff this partition is not a root partition.
+// Returns the fetched data, if fetching was necessary and successful.
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartition(ctx context.Context) (*persistencespb.VersioningData, error) {
```

Comment:
I would like this even more if the fetches/invalidates followed the existing tree structure, to reduce fan-out/fan-in. i.e. don't fetch from the root, fetch from the parent (which might fetch from its parent, etc.). and the root shouldn't invalidate everyone, it should invalidate its children.

we can put it off until later though.